libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,759 @@
+    .pRootCa           = NULL,
+    .rootCaSize        = 0,
+    .sendTimeoutMs     = TEST_TRANSPORT_SND_TIMEOUT_MS,
+    .recvTimeoutMs     = TEST_TRANSPORT_RCV_TIMEOUT_MS
+};
+
+static uint8_t networkBuffer[ BUFFER_LEN ] = { 0 };
+static Socket_t mocTcpSocket = ( Socket_t ) MOCT_TCP_SOCKET;
+static NetworkContext_t networkContext = { 0 };
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+    networkContext.tcpSocket = mocTcpSocket;
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails with NULL parameters
+ * and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Invalid_Params( void )
+{
+    TransportSocketStatus_t returnStatus;
+    ServerInfo_t invalidServerInfo = { 0 };
+
+    invalidServerInfo.port = PORT;
+    returnStatus = SecureSocketsTransport_Connect( NULL,
+                                                   &serverInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   NULL,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   NULL );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    invalidServerInfo.pHostName = NULL;
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    invalidServerInfo.pHostName = INVALID_HOSTNAME;
+    invalidServerInfo.hostNameLength = 0;
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    invalidServerInfo.hostNameLength = strlen( INVALID_HOSTNAME );
+
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
```

Comment:
It seems like `SOCKETS_Close` is called in the current implementation for the invalid input parameter case when the host name length is long. `SOCKETS_Close` shouldn't necessarily be called in any case of an invalid input parameter.
Could you update the implementation of `SecureSocketsTransport_Connect` to not call `SOCKETS_Close` for an invalid input parameter?

Reply:
fixed in f5095e0ad5300ba6f97a03cd3d1622b525afc0b9